internal/services/azurestackhci/stack_hci_cluster_resource.go

```diff
@@ -308,3 +345,33 @@ func resourceArmStackHCIClusterDelete(d *pluginsdk.ResourceData, meta interface{
 
 	return nil
 }
+
+func expandSystemAssigned(input []interface{}) (*identity.SystemAndUserAssignedMap, error) {
+	if len(input) == 0 || input[0] == nil {
+		return &identity.SystemAndUserAssignedMap{
+			Type: identity.TypeNone,
+		}, nil
+	}
+
+	return &identity.SystemAndUserAssignedMap{
+		Type: identity.TypeSystemAssigned,
+	}, nil
+}
+
+func flattenSystemAssigned(input *identity.SystemAndUserAssignedMap) []interface{} {
+	if input == nil {
+		return []interface{}{}
+	}
+
+	if input.Type == identity.TypeNone {
+		return []interface{}{}
+	}
+
+	return []interface{}{
+		map[string]interface{}{
+			"type":         input.Type,
+			"principal_id": input.PrincipalId,
+			"tenant_id":    input.TenantId,
+		},
+	}
+}
```

Comment:
Same question here, why can't we use the expand and flatten functions in the `identity` package?

Reply:
The service API only supports systemAssigned Identity, which means there should be no `identity_ids` field in the tf schema, but the swagger and SDK uses `identity.SystemAndUserAssignedMap`, so need mannually skip the `identity_ids` in expand/flatten function.